lib/matplotlib/collections.py

```diff
@@ -293,6 +294,12 @@ def draw(self, renderer):
 
         if self._hatch:
             gc.set_hatch(self._hatch)
+            try:
+                gc.set_hatch_color(self._hatch_color)
+            except AttributeError:
+                # if we end up with a GC that does not have this method
+                warnings.warn("Your backend does not support setting the "
```

Comment:
Could you define `set_hatch_color` in the base graphicscontext class as a noop that prints the warning?  (would avoid the repetition below).